src/transaction.ts

```diff
@@ -79,7 +92,21 @@ class Transaction extends DatastoreRequest {
     this.id = options.id;
     this.readOnly = options.readOnly === true;
 
-    this.request = datastore.request_.bind(datastore);
+    this.request_ = datastore.request_;
+
+    this.get = datastore.get;
+
+    this.runQuery = datastore.runQuery;
+
+    this.runQueryStream = datastore.runQueryStream;
+
+    this.createReadStream = datastore.createReadStream;
+
+    this.insert = datastore.insert.bind(this);
+
+    this.update = datastore.update.bind(this);
+
+    this.upsert = datastore.upsert.bind(this);
```

Comment:
1. The core difference in `datastore.merge()` vs `transaction.merge()` methods.
2. If `Transaction` inherits from `Request` it also inherits type signatures for methods `merge`, `insert`, `update`, `upsert`, `save`. Typescript will not allow overriding child class method's signature in a way that it is incompatible with parent class method's signature.
    - `transaction.merge()` is introduced to accept two parameters `currentEntity` and `entityToMerge`.
  Since workflow would be `transaction.get()` --then--> `transaction.merge()` with something else. 
    - `datastore.merge()` still takes `Entity | Entity[]` and utilizes `transaction.merge()` for each of the `Entity` provided.
    - `merge`, `insert`, `update`, `upsert`, `save` now have proper `void` return signatures.
That resolves an issue described in the snippet below ([link](https://github.com/googleapis/nodejs-datastore/issues/540#issue-523663251))

      > ![Screen Shot 2020-04-03 at 12 15 33 PM](https://user-images.githubusercontent.com/7764119/78390235-3a3e2080-75b2-11ea-9b37-3955e2185127.png)

Reply:
Sorry I missed this! Does the Transaction class still get documentation for these methods that are bound on the instance? Params, examples, etc?

> Typescript will not allow overriding child class method's signature in a way that it is incompatible with parent class method's signature.

Is this a bug? Is there a hacky solution? As a workaround, can we change one or both signatures to find a happy medium? `any` to the rescue somehow? Rename a method to something unique, e.g. `mergeMultiple`? No annotations for this method, instead just link to the docs?

Overall, I still wish there was another way around this, so anything above sounds better than getting rid of the code that was using built-in JS structure. Maybe insignificant, but we're losing performance by binding methods in the constructor, as opposed to defined-once, prototypal inheritance.